### DISCUSSION TITLE
What's in an Intersection?

### DISCUSSION THREAD

**Discussion by BudgieInWA** [a-b-street/osm2streets#191] [General]
I would like to get clear on what sort of features our intersection abstraction(s) ought to have. It is a pretty important one, having meaning to the road users (like "don't change lanes in an intersection", "don't even enter the intersection until your exit is clear").

to get started, I would want it to have things like
- a perimeter that we can walk with lanes in order,
- all the legal turn paths.
- each arms' relationship to the intersection: (stop/give way line, continues in, exit only, ...)
- all the pedestrian traffic crossings running through it
- ...

**Comment by dabreegster** [comment]
Offhand I can think of a few more elements...

- https://en.wikipedia.org/wiki/Advanced_stop_line, maybe arguably not part of the intersection
- https://nacto.org/publication/dont-give-up-at-the-intersection/protected-intersections/ lists more elements
- Chapter 10 of https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/951074/cycle-infrastructure-design-ltn-1-20.pdf has a taxonomy of ped/cycle crossings

**Comment by BudgieInWA** [comment]
<details>
<summary>Some random figure of intersections I've found that show pertinent features drawn in</summary>
![image](https://user-images.githubusercontent.com/3784591/190323154-67a2b4e0-ec4a-4308-bd1d-8667d79092ce.png)
![image](https://user-images.githubusercontent.com/3784591/190323824-701af331-4864-4626-a88e-a3d9ecffc35f.png)
![image](https://user-images.githubusercontent.com/3784591/190324276-b3311f96-6c5b-4ab2-af85-be30ede397e9.png)
</details>

**Comment by dabreegster** [comment]
The formatting broke.. trying to copy the links:

![image](https://user-images.githubusercontent.com/3784591/190323154-67a2b4e0-ec4a-4308-bd1d-8667d79092ce.png)
![image](https://user-images.githubusercontent.com/3784591/190323824-701af331-4864-4626-a88e-a3d9ecffc35f.png) 

![image](https://user-images.githubusercontent.com/3784591/190324276-b3311f96-6c5b-4ab2-af85-be30ede397e9.png)
